### DISCUSSION TITLE
Are you allowed to delete tombstone events from mt_events?

### DISCUSSION THREAD

**Discussion by jclijmans** [JasperFx/marten#3390] [Q&A]
Due to a sub-optimal application, we had lots of concurrency errors resulting in lots of tombstone events in the mt_events table.

Can we delete those tombstone events without causing issues in our application?
I could not find any guidance regarding this in the documentation.

**Comment by mysticmind** [comment]
I think you had asked the same question in Discord. 

Adding an answer for anyone who is searching here. Tombstone events are a record of what has happened so deleting them shouldn't cause issues. In production scenario, always double check with a test and then attempt to delete the tombstone events.

It is a good idea to add some documentation on this, will look into it.

**Comment by jeremydmiller** [comment]
If the high water mark in mt_event_progression, it does no harm to delete tombstone events
